web3/contract/__init__.py

```diff
@@ -0,0 +1,10 @@
+from web3.contract.async_contract import (  # noqa: F401
+    AsyncContract,
+    AsyncContractCaller,
+)
+
+from web3.contract.contract import (  # noqa: F401
+    Contract,
+    ContractConstructor,
+    ContractCaller,
+)
```

Comment:
These are the only classes that get exported to other files in our codebase. I _think_ that our public API all stems from either `Contract` or `AsyncContract`, but I wonder if it's worth exposing other classes like `ContractFunctions` or `ContractEvents` here too?

Reply:
I would lean towards only exposing what is necessary and add more if when it's called for. Adding things here wouldn't be breaking, right?